### DISCUSSION TITLE
[Authentication] Use OpenIddict Client instead of IdentityModel

### DISCUSSION THREAD

**Discussion by xperiandri** [unoplatform/uno.extensions#1883] [Ideas]
## What would you like to be added:

IdentityModel is a petty primitive library.
Contrary OpenIddict Client is a secure sophisticated library that covers all the possible scenarios (except WASM)

## Why is this needed:

Allows to easily perform authentication setup including usage of DI

## For which Platform:

- iOS
- Android
- Windows
- macOS
- Linux

**Comment by nickrandolph** [comment]
Happy to consider this as an alternative - the Authentication model is open enough that it should be easy enough to add OpenIddict Client

**Reply by xperiandri** [reply]
@jeromelaban @nick_randolph I succeeded with use of OpenIddict Client (on WinUI for now)
https://github.com/xperiandri/openiddict-core/tree/uno
So I propose to Freeze Uno.Extensions.Authentication.OIDC and invest in it instead.
As Uno.Extensions.Authentication.OIDC is very closed and inextensible contrary to OpenIddict Client.
Step left or right from the sample and you cannot do anything custom with Uno.Extensions.Authentication.OIDC

**Reply by DevTKSS** [reply]
@xperiandri could you think of adding that as a sample to Uno.Samples like @nickrandolph asked?
I tryed the links you send on the other Discussion (not finding it at this moment) and its locally here only failing on this one Package: `<PackageVersion Include="OpenIddict.Client.SystemIntegration" Version="7.0.0" />` with a version conflict with the Uno used ones for the Google/Android things.

```
Warnings as Errors: A Downgrade of the Package "Xamarin.AndroidX.Browser" from 1.8.0.10 to 1.8.0.8 has been discovered. To select a different version, reference the package directly from the project.
 DevTKSS.MyManufacturerERP -> OpenIddict.Client.UnoIntegrations -> OpenIddict.Client.SystemIntegration 7.0.0 -> Xamarin.AndroidX.Browser (>= 1.8.0.10) 
 DevTKSS.MyManufacturerERP -> Xamarin.AndroidX.Browser (>= 1.8.0.8)
```

**Comment by nickrandolph** [comment]
For now this has been moved to a discussion. Rather than bulk out Uno.Extensions Authentication with more providers, I'd suggest adding a sample to uno.samples repository on how OpenIddict Client can be used
